### DISCUSSION TITLE
Is there a way to clear memory?

### DISCUSSION THREAD

**Discussion by JoniJnm** [spatie/image#275] [General]
Hi, checking the [GdDriver](https://github.com/spatie/image/blob/main/src/Drivers/Gd/GdDriver.php) I don't see a function calling `imagedestroy`. I don't see neither a "destroy" function or something similar in [Image class](https://github.com/spatie/image/blob/main/src/Image.php).

Is PHP internally calling `imagedestroy()` when the reference to the instance of Image is lost during the gc cycle check?

In the [imagedestroy](https://www.php.net/manual/es/function.imagedestroy.php). page, the comments say that PHP clears the resource when the script ends.

Thanks.

**Comment by misakstvanu** [comment]
Spanish documentation unfortunately doesn't state this, but imagedestroy doesn't have any effect since 8.0. All [resources](https://www.php.net/manual/en/language.types.resource.php) in modern PHP are freed when there are no more references to them.

See [https://www.php.net/manual/en/function.imagedestroy.php](https://www.php.net/manual/en/function.imagedestroy.php)

**Reply by JoniJnm** [reply]
Oh, I didn't realize I posted the spanish version.

Thanks!
